raft/src/test/java/org/apache/kafka/raft/KafkaRaftClientTest.java

```diff
@@ -903,6 +907,111 @@ public void testInvalidVoteRequest() throws Exception {
         assertFailedVoteResponse(Errors.INVALID_REQUEST);
     }
 
+    @Test
+    public void testPurgatoryFetchTimeout() throws Exception {
+        int otherNodeId = 1;
+        int epoch = 5;
+
+        Set<Integer> voters = Utils.mkSet(localId, otherNodeId);
+        KafkaRaftClient client = initializeAsLeader(voters, epoch);
+
+        // Follower sends a fetch which cannot be satisfied immediately
+        int maxWaitTimeMs = 500;
+        FetchQuorumRecordsRequestData fetchRequest = fetchQuorumRecordsRequest(
+            epoch, otherNodeId, 1L, epoch, maxWaitTimeMs);
+        channel.mockReceive(new RaftRequest.Inbound(channel.newCorrelationId(), fetchRequest, time.milliseconds()));
+        client.poll();
+        assertEquals(0, channel.drainSendQueue().size());
+
+        // After expiration of the max wait time, the fetch returns
+        time.sleep(maxWaitTimeMs);
+        client.poll();
+        assertFetchQuorumRecordsResponse(epoch, localId);
+    }
+
+    @Test
+    public void testPurgatoryFetchSatisfiedByWrite() throws Exception {
+        int otherNodeId = 1;
+        int epoch = 5;
+
+        Set<Integer> voters = Utils.mkSet(localId, otherNodeId);
+        KafkaRaftClient client = initializeAsLeader(voters, epoch);
+
+        // Follower sends a fetch which cannot be satisfied immediately
+        FetchQuorumRecordsRequestData fetchRequest = fetchQuorumRecordsRequest(
+            epoch, otherNodeId, 1L, epoch, 500);
+        channel.mockReceive(new RaftRequest.Inbound(channel.newCorrelationId(), fetchRequest, time.milliseconds()));
+        client.poll();
+        assertEquals(0, channel.drainSendQueue().size());
```

Comment:
nit: could check empty.

Reply:
Using `assertEquals` with the size provides more information in the case of failure.